share/module_check_a_mundo.F

```diff
@@ -194,6 +194,28 @@ SUBROUTINE check_nml_consistency
          count_fatal_error = count_fatal_error + 1
          END IF
       ENDDO
+!-----------------------------------------------------------------------
+! Check that Deng Shallow Convection Must work with MYJ or MYNN PBL
+!-----------------------------------------------------------------------
+      DO i = 1, model_config_rec % max_dom
+         IF ( model_config_rec % shcu_physics(i) == dengshcuscheme .AND. &
+              (model_config_rec % bl_pbl_physics(i) /= myjpblscheme .OR. &
+               model_config_rec % bl_pbl_physics(i) /= mynnpblscheme2 ) ) THEN
+            wrf_err_message = '--- ERROR: Deng shallow convection can only work with MYJ or MYNN PBL '
+            CALL wrf_message ( wrf_err_message )
+            wrf_err_message = '--- ERROR: Fix shcu_physics and bl_pbl_physics in namelist.input.'
+         CALL wrf_debug ( 0, TRIM( wrf_err_message ) )
+         count_fatal_error = count_fatal_error + 1
+         END IF
+         IF ( model_config_rec % shcu_physics(i) == dengshcuscheme .AND. &
+              model_config_rec % cu_physics(i) /= nocuscheme  ) THEN
+            wrf_err_message = '--- ERROR: Deng shallow convection must run with cu_physics=0 '
+            CALL wrf_message ( wrf_err_message )
+            wrf_err_message = '--- ERROR: Fix shcu_physics and cu_physics in namelist.input.'
```

Comment:
@smileMchen The same here: "Fix shcu_physics or cu_physics in namelist.input". They don't need to change both.